yang/frr-zebra.yang

```diff
@@ -0,0 +1,1887 @@
+    }
+    leaf debug-rib {
+      type empty;
+      description "Debug RIB processing.";
+    }
+    leaf debug-rib-detail {
+      type empty;
+      description "Debug RIB processing details.";
+    }
+    leaf debug-fpm {
+      type empty;
+      description "Debug the FIB Push Interface subsystem.";
+    }
+    leaf debug-nht {
+      type empty;
+      description "Debug Nexthop-tracking.";
+    }
+    leaf debug-nht-detail {
+      type empty;
+      description "Debug Nexthop-tracking details.";
+    }
+    leaf debug-mpls {
+      type empty;
+      description "Debug MPLS.";
+    }
+    leaf debug-vxlan {
+      type empty;
+      description "Debug VxLAN.";
+    }
+    leaf debug-pw {
+      type empty;
+      description "Debug pseudowires.";
+    }
+    leaf debug-dplane {
+      type empty;
+      description "Debug the dataplane subsystem.";
+    }
+    leaf debug-dplane-detail {
+      type empty;
+      description "Debug dataplane subsystem details.";
+    }
+    leaf debug-mlag {
+      type empty;
+      description "Debug MLAG.";
+    }
+  }
+
+  /*
+   * Main zebra container
+   */ 
+  container zebra {
+    description
+      "Data model for the Zebra daemon.";
+
+    leaf mcast-rpf-lookup {
+      type frr-zebra:mcast-rpf-lookup-mode;
+      default "mrib-then-urib";
+      description
+	"Multicast RPF lookup behavior.";
+    }
+
+    leaf ip-forwarding {
+      type boolean;
+      description
+	"IP forwarding status.";
+    }
+
+    leaf ipv6-forwarding {
+      type enumeration {
+	enum "unknown" {
+	  value -1;
+	  description
+	    "Unknown state.";
+	}
+	enum "off" {
+	  value 0;
+	  description
+	    "IPv6 forwarding disabled.";
+	}
+	enum "on" {
+	  value 1;
+	  description
+	    "IPv6 forwarding enabled.";
+	}
+      }
+      description
+	"IPv6 forwarding status.";
+    }
+
+    leaf workqueue-hold-timer {
+      type uint32 {
+        range "0..10000";
+      }
+      units "milliseconds";
+      default "10";
+      description
+        "Work-queue processing hold timer, in milliseconds.";
+    }
+
+    leaf zapi-packets {
+      type uint32 {
+        range "1..10000";
+      }
+      default "1000";
+      description
+        "Number of ZAPI packets to process before relinquishing
+        the main thread.";
+    }
+
+    container import-kernel-table {
+      description
+        "Parameters to use when importing IPv4 routes from a non-main kernel
+        routing table.";
+
+      leaf table-id {
+        type uint32 {
+          range "1..252";
+        }
+        description
+          "The kernel table id.";
+      }
+      leaf distance {
+        type uint32 {
+          range "1..255";
+        }
+        default "15";
+        description
+          "The admin distance to use for imported routes.";
+      }
+      leaf route-map {
+        type string;
+        description
+          "A route-map to filter imported routes.";
+      }
+    }
+
+    leaf allow-external-route-update {
```

Comment:
How about using `boolean` here? It is either configured or not. Again, the only difference I see here is that when using `empty` the user will have to "create" this leaf, with `boolean` he could just "modify" it.

Reply:
yes, right - but if it's not configured, do we need a default value? I guess it's not clear to me whether it's better to handle a 'create' or a 'modify' operation.